### PR TITLE
2281081 - Port "Data value is empty (data_is_empty)" condition to 8.x

### PR DESCRIPTION
**Pull request by paranojik**
This is basically a copy of https://github.com/fago/rules/pull/66 with added unit test.
Related d.g.o. issue: https://www.drupal.org/node/2281081.

### CODE REVIEW

File: tests/src/Condition/DataIsEmptyTest.php [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  /**
+   * Tests context value setting and getting.
+   *
+   * @covers ::setContextValue()
+   * @covers ::getContextValue()
+   */
+  public function testContextValue() {
+    // We can't mock the NodeInterface because there is a bug in PHPUnit below
+    // version 3.8 that causes mocking of interfaces that extend \Traversable
+    // to fail. @see https://github.com/sebastianbergmann/phpunit-mock-objects/issues/103
```
Comment (klausi) [inline tests/src/Condition/DataIsEmptyTest.php]: Drupal core now uses PHPunit 4, so we should be able to use the interface now. Please check.

**Comment by paranojik** [comment]
Yup. Works just fine. Thanks, @klausi!

**Comment by klausi** [comment]
The context stuff just landed in core and @fubhy updated all the condition plugins, so you will have to update the annotation, too (and remove the static context method).

**Comment by paranojik** [comment]
Context stuff updated...

**Pull request merged**